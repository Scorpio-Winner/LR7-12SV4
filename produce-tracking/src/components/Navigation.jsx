import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { Box, Link, Toolbar } from '@mui/material';

const Navigation = () => {
  const [role, setRole] = useState(localStorage.getItem("role"));


  const handleLogout = () => {
    localStorage.removeItem("token");
    localStorage.removeItem("role");
    window.location.reload();
  };


  return (
    <Toolbar>
      <Box sx={{ flexGrow: 1 }}>
        <Link
          component={RouterLink}
          to="/orders"
          color="black"
          underline="none"
          sx={{
            marginRight: 2,
            '&:hover': {
              textDecoration: 'underline',
            },
            fontSize: '1.2rem',
          }}
        >
          All orders
        </Link>
        {role === 'admin' && (
          <Link
            component={RouterLink}
            to="/orders/create"
            color="black"
            underline="none"
            sx={{
              marginRight: 2,
              '&:hover': {
                textDecoration: 'underline',
              },
              fontSize: '1.2rem',
            }}
          >
            Create order
          </Link>
        )}
        <Link
          component={RouterLink}
          to="/report"
          color="black"
          underline="none"
          sx={{
            marginRight: 2,
            '&:hover': {
              textDecoration: 'underline',
            },
            fontSize: '1.2rem',
          }}
        >
          Report
        </Link>
        <Link
          color="black"
          underline="none"
          onClick={handleLogout}
          sx={{
            marginRight: 2,
            '&:hover': {
              textDecoration: 'underline',
            },
            fontSize: '1.2rem',
          }}
        >
          Logout
        </Link>
      </Box>
    </Toolbar>
  );
};

export default Navigation;