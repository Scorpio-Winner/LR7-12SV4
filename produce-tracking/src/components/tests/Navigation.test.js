import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation';

describe('Navigation', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('renders correctly for admin role', () => {
    localStorage.setItem('role', 'admin');
    const tree = renderer
      .create(
        <Router>
          <Navigation />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for non-admin role', () => {
    localStorage.setItem('role', 'user');
    const tree = renderer
      .create(
        <Router>
          <Navigation />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when role is not set', () => {
    const tree = renderer
      .create(
        <Router>
          <Navigation />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when role is null', () => {
    localStorage.setItem('role', null);
    const tree = renderer
      .create(
        <Router>
          <Navigation />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
