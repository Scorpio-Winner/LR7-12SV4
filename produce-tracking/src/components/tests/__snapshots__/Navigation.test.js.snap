// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly for admin role 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/orders"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      All orders
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/orders/create"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Create order
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/report"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Report
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Navigation renders correctly for non-admin role 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/orders"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      All orders
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/report"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Report
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Navigation renders correctly when role is not set 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/orders"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      All orders
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/report"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Report
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Navigation renders correctly when role is null 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/orders"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      All orders
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      href="/report"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Report
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;
